### PR TITLE
Add form to tag Education content to beta taxonomy

### PR DESCRIPTION
**Pull request by Davidslv**
For documents belonging to DfE and SFA, show a button to tag to the
Finding Things education taxonomy. This allows stakeholders to keep
tagging content to the taxonomy during beta.

## Look & Feel

screen shot 2017-01-12 at 15 50 15
screen shot 2017-01-12 at 15 50 12
screen shot 2017-01-12 at 15 49 56

Trello:
https://trello.com/c/miTSiYfk/409-create-tagging-interface-for-new-taxonomy-in-whitehall

Paired with @MatMoore and @klssmith from @alphagov/team-gov-uk-finding-things

### CODE REVIEW

**Comment by tijmenb** [comment]
This looks really good!

**Comment by MatMoore** [comment]
I moved the mixin from `Editions` to the various models that include the `Edition::Organisations` concern, because not every `Edition` has the organisations association.

As far as I can tell, the only exception is `CorporateInformationPage`, which isn't relevant to the beta.

File: app/controllers/admin/edition_tags_controller.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+class Admin::EditionTagsController < Admin::BaseController
+  before_filter :find_edition
+  before_filter :enforce_permissions!
+  before_filter :limit_edition_access!
+
+  def enforce_permissions!
```
Comment (tijmenb) [inline app/controllers/admin/edition_tags_controller.rb]: It's best to make this method private (Rails does magic things to public methods on controllers).

File: app/models/edition_tag_form.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+class EditionTagForm
```
Comment (tijmenb) [inline app/models/edition_tag_form.rb]: Is it worth being explicit here by calling it `EditionTaxonomyTagForm` or something like that? (I'm fine either way)

File: app/models/edition/taggable_organisations.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+module Edition::TaggableOrganisations
+  extend ActiveSupport::Concern
+
+  def can_be_tagged_to_taxonomy?
+    organisations_content_ids = organisations.map(&:content_id)
+
+    has_taggable_organisations?(organisations_content_ids)
```
Comment (tijmenb) [inline app/models/edition/taggable_organisations.rb]: Could we come up with a better name for `taggable_organisations`? The organisations' documents are taggable, not the org itself.

File: app/controllers/admin/edition_tags_controller.rb (hunk 1 above)
Comment (Davidslv) [inline app/controllers/admin/edition_tags_controller.rb]: Will change it, also will do another PR to make all of them private methods (seems to be a mix between public/private in different controllers)

**Comment by Davidslv** [comment]
This is missing a feature flag as discussed offline

**Comment by Davidslv** [comment]
This can only be deployed after implementation of feature flag and the deploy of puppet with the new environment variable (feature flag) https://github.com/alphagov/govuk-puppet/pull/5361
